### ISSUE TITLE
Unexpected html structure when assigning a certain string content to EditorOptions.initialContent

### ISSUE THREAD

**Issue by Dream0827** [microsoft/roosterjs#1423]
**Describe the bug**
When assigning specificized content to `EditorOptions.initialContent`, the rendering DOM structure will be different from the initial content.

**To Reproduce**
1. Set `EditorOptions.initialContent` value as below html content:
```
<div>
    <a href="https://microsoft.sharepoint.com/video.mp4">
       Test link
       <div>Block element</div>
    </a>
</div>
```
2. Verify the rendered Html dom structure

**Expected behavior**
Same as the initial content

**Screenshots**
The Html content added an extra empty anchor node.
![image](https://user-images.githubusercontent.com/118722930/203009821-e8e7c23b-1ecf-46da-a008-be6caa37769e.png)

**Device Information**
 - Browser: chrome
 - Version: 8.36.0

**Additional context**
If adding the above content twice and setting it as initial content, the second will keep consistent with the initial content.
![image](https://user-images.githubusercontent.com/118722930/203010866-8d6f0ec7-1d13-47b6-926e-511412132fab.png)

**Comment by JiuqingSong** [comment]
It doesn't repro to me.

Here is my test: https://codepen.io/songthin/pen/XWYEVQw

![image](https://user-images.githubusercontent.com/23065085/203611387-8bc76fd0-6d1f-403d-a75e-8feb9ce2fa69.png)

**Comment by Dream0827** [comment]
Is the result by design? 
Why are the anchor containing the "Test link" text and the anchor containing the div "Block element" not put in a div, but separated into two Divs?
![image](https://user-images.githubusercontent.com/118722930/203677396-54e9f6cb-1180-4870-9493-e18edfd3ffa9.png)

If the above result is by design, I tested that if there are two same anchor nodes "Test link 1" and "Test link 2" and they both include a child div node "Block element 1" and "Block element 2", such as the below initial content, **why the result of "Test link 1" anchor node and "Test link 2" anchor node are different?**
<img width="534" alt="image" src="https://user-images.githubusercontent.com/118722930/203678268-e3eb3652-f081-4535-b90a-2d5a222d2d37.png">
![image](https://user-images.githubusercontent.com/118722930/203678600-0f2e3549-8dba-48d0-93b4-1dd92f6d063d.png)

**Comment by JiuqingSong** [comment]
It sounds like by design to me. It is a result of TypeInContainerPlugin which will always make sure there is a block element around current cursor. In your case, there is a anchor element but no block element (DIV) at the first line. So the code will split the A tag and add a DIV tag.  I feel it is a little weird to have A tag outside DIV. Is there any bad impact to split the A tag other than change the DOM structure?

**Comment by Dream0827** [comment]
When the anchor node only has a div child node, if splitting the A tag to a separate div, there will be an accessibility issue. It will read the empty anchor node.

**Comment by JiuqingSong** [comment]
If after split an empty A tag is created (as your screenshot shows), it could be a problem. However, I'm not able to repro it. Can you try with https://codepen.io/songthin/pen/XWYEVQw and see if the same issue still happen there? If not, can you build a real repro environment so that I can debug?

**Comment by JiuqingSong** [comment]
This is what I see from my test:
<img width="354" alt="image" src="https://user-images.githubusercontent.com/23065085/204076634-f37af71c-f348-40c3-ac25-4ca975b823df.png">
There is no empty A tag.

If you want this to be fixed, a live repro is required. Otherwise even if we make some fix how can we tell if that change can really fix the issue?

**Comment by Dream0827** [comment]
If you remove the "Test link" text from the A tag, then there will be an empty A tag. 
Here is the repo test: https://codepen.io/dreamxcf/pen/RwJJWoM
<img width="510" alt="image" src="https://user-images.githubusercontent.com/118722930/204088653-04b81bc4-329f-428e-b6a2-cfd750ba8540.png">

**Comment by JiuqingSong** [comment]
This will be solved in #1553
